lib/mongodb.js

```diff
@@ -180,6 +188,21 @@ MongoDB.prototype.fromDatabase = function (model, data) {
         // Convert the Binary into String
         data[p] = data[p].toString();
       }
+    }  // check if prop contains EmbedsMany
+    else if(prop && prop.type.constructor === Array &&
```

Comment:
@raymondfeng : I implemented the method ( 3d5b38292aa42dfed6a4e5f5bca68cfa53329ea6 ), now this check if the property corresponds to a relation in the model:

```javascript
/*!
 * Convert the data from database to JSON
 *
 * @param {String|Object} model The model name or the model class
 * @param {Object} data The data from DB
 */
MongoDB.prototype.fromDatabase = function (model, data) {
  var self = this;
  var props;
  var settings;
  var relations = [];
  if (!data) {
    return null;
  }
  if(typeof model === 'string'){
    model = this._models[model];
  }
  props = model.properties;
  settings = model.settings;
  if(settings){
    for(var k in settings.relations){
      if(settings.relations[k].property){
        relations[settings.relations[k].property] = settings.relations[k];
      }
    }
  }
  for (var p in props) {
    var prop = props[p];
    if (prop && prop.type === Buffer) {
      if(data[p] instanceof mongodb.Binary) {
        // Convert the Binary into Buffer
        data[p] = data[p].read(0, data[p].length());
      }
    } else if(prop && prop.type === String) {
      if(data[p] instanceof mongodb.Binary) {
        // Convert the Binary into String
        data[p] = data[p].toString();
      }
    } // check if prop is a relation EmbedsMany
    else if(prop && relations[p] && relations[p].type === 'embedsMany'){
      for(var i = 0; i < data[p].length; i++){
        data[p][i] = self.fromDatabase({
          model: prop.type[0],
          properties: prop.type[0].definition.properties,
          settings: prop.type[0].definition.settings
        },data[p][i]);
      }
    } // check if prop is a relation EmbedsOne
    else if(prop && relations[p] && relations[p].type === 'embedsOne'){
      data[p] = self.fromDatabase({
        model: prop.type,
        properties: prop.type.definition.properties,
        settings: prop.type.definition.settings
      },data[p]);
    }
  }
  return data;
};
```